webroot/src/pages/LicensingDetail/LicensingDetail.ts

```diff
@@ -19,12 +21,66 @@ export default class LicensingDetail extends Vue {
     //
     // Lifecycle
     //
+    async mounted() {
+        await this.fetchLicenseeData();
+
+        if (!this.licensee) {
+            this.$router.push({ name: '404' });
+        }
+    }
 
     //
     // Computed
     //
+    get licenseStore(): any {
+        return this.$store.state.license;
+    }
+
+    get licenseeId(): string {
+        const routeId: string = this.$route.params.licenseeId as string || '';
+
+        return routeId;
+    }
+
+    get licensee(): Licensee | null {
+        const { licenseeId } = this;
+        let storeRecord: Licensee | null = null;
+
+        if (licenseeId && this.licenseStore.model) {
+            storeRecord = this.$store.getters['license/licenseeById'](licenseeId);
+        }
+
+        return storeRecord;
+    }
```

Comment:
This might be one that I'd prefer to leave closer to as-is. In general, our team is going to prefer readability / self-documenting over clever one-liners. Open to discussion if you feel strongly.

Reply:
That's fine with me. This was certainly the one where I thought to myself that it might be worth not making this a one-liner.